app/test/block_production_test.go

```diff
@@ -55,3 +57,35 @@ func (s *BlockProductionTestSuite) Test_BlockOneTransactionNonInclusion() {
 	require.Error(err) // change this to require.NoError(err) to see the error
 	require.ErrorContains(err, "rpc error: code = Unknown desc = codespace sdk code 18: invalid request: failed to load state at height 0; no commit info found (latest height: 0)")
 }
+
+// Test_FirstBlockIsEmpty tests whether the first block is empty.
+func (s *BlockProductionTestSuite) Test_FirstBlockIsEmpty() {
+	require := s.Require()
+	// wait until height 1 before posting transactions
+	// otherwise tx submission will fail
+	time.Sleep(1 * s.timeoutCommit)
```

Comment:
[question][not blocking] it's not immediately obvious what the relationship is between timeout commit and block height 1. If this only cares about waiting for block height 1 then it could be

```suggestion
	// Wait until block height 1 before posting transactions otherwise tx
	// submission will fail.
	_, err := s.cctx.WaitForHeightWithTimeout(1, s.timeoutCommit)
	require.NoError(err)
```

Reply:
Both methods work well: either sleeping for the timeoutCommit or waiting for height 1. However, for a clearer understanding, I've chosen to explicitly wait, and I've incorporated your suggestion in this [PR](https://github.com/celestiaorg/celestia-app/pull/2284). Thanks.